scrapy/extensions/logstats_extended.py

```diff
@@ -0,0 +1,132 @@
+import logging
+from datetime import datetime
+
+from twisted.internet import task
+
+from scrapy import signals
+from scrapy.exceptions import NotConfigured
+from scrapy.utils.serialize import ScrapyJSONEncoder
+
+logger = logging.getLogger(__name__)
+
+
+class LogStatsExtended:
+    """Log basic scraping stats periodically"""
+
+    def __init__(
+            self, stats, interval=60.0,
+            ext_stats_enabled=False,
+            ext_stats_include=None,
+            ext_stats_exclude=None,
+            ext_delta_enabled=False,
+            ext_delta_include=None,
+            ext_delta_exclude=None,
+            ext_timing_enabled=False,
+            ext_timing_include=None,
+            ext_timing_exclude=None,
+    ):
+        self.stats = stats
+        self.interval = interval
+        self.multiplier = 60.0 / self.interval
+        self.task = None
+        self.encoder = ScrapyJSONEncoder(sort_keys=True, indent=4)
+        self.ext_stats_enabled = ext_stats_enabled
+        self.ext_stats_include = ext_stats_include
+        self.ext_stats_exclude = ext_stats_exclude
+        self.ext_delta_enabled = ext_delta_enabled
+        self.ext_delta_include = ext_delta_include
+        self.ext_delta_exclude = ext_delta_exclude
+        self.ext_timing_enabled = ext_timing_enabled
+        self.ext_timing_include = ext_timing_include
+        self.ext_timing_exclude = ext_timing_exclude
+
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        interval = crawler.settings.getfloat("LOGSTATS_INTERVAL")
+        ext_stats_enabled = crawler.settings.getbool("LOGSTATS_EXT_STATS_ENABLED")
+        ext_stats_include = crawler.settings.getlist("LOGSTATS_EXT_STATS_INCLUDE", [])
+        ext_stats_exclude = crawler.settings.getlist("LOGSTATS_EXT_STATS_EXCLUDE", [])
+        ext_delta_enabled = crawler.settings.getbool("LOGSTATS_EXT_DELTA_ENABLED")
+        ext_delta_include = crawler.settings.getlist("LOGSTATS_EXT_DELTA_INCLUDE", [])
+        ext_delta_exclude = crawler.settings.getlist("LOGSTATS_EXT_DELTA_EXCLUDE", [])
+        ext_timing_enabled = crawler.settings.getbool("LOGSTATS_EXT_TIMING_ENABLED")
```

Comment:
I think we can keep `LOGSTATS_` as the prefix as long as we make sure there are no conflicts with existing settings:

```suggestion
        ext_stats_include = crawler.settings.getlist("LOGSTATS_INCLUDE", [])
        ext_stats_exclude = crawler.settings.getlist("LOGSTATS_EXCLUDE", [])
        ext_delta_include = crawler.settings.getlist("LOGSTATS_DELTA_INCLUDE", [])
        ext_delta_exclude = crawler.settings.getlist("LOGSTATS_DELTA_EXCLUDE", [])
        ext_timing_enabled = crawler.settings.getbool("LOGSTATS_EXTENDED_TIME")
```

We could also modify the existing extension to be smart about settings specific to this new extension, e.g.
- Raise `NotConfigured` if `LOGSTATS_EXTENDED` is `True`.
- Log a warning if one of these 5 settings are defined with a non-default value, indicating that they only have effect if `LOGSTATS_EXTENDED` is set to `True`.

Reply:
> Log a warning if one of these 5 settings are defined with a non-default value, indicating that they only have effect if LOGSTATS_EXTENDED is set to True.

To implement this I'd prefer to use existing `overridden settings` feature in scrapy.

If we define `PERIODIC_LOG_STATS=None; PERIODIC_LOG_DELTA=None; PERIODIC_LOG_TIMING_ENABLED=None` in scrapy/settings/default_settings.py - on run with.. non default values of this settings - it will be reflected in overridden settings log entry